### PR TITLE
Fix function with multiple namespaces

### PR DESCRIPTION
**Pull request by VincentLanglet**
Hi @theofidry, I discovered a bug with the php-scoper.

Before the fix, I get
```
namespace Acme\Emca {
    if (!function_exists('Acme\Emca\foo')) { function Emca\foo() { return \Humbug\Acme\Emca\foo(...func_get_args()); } }
}
```
which is crashing.

### CODE REVIEW

**Comment by theofidry** [comment]
Thank you @VincentLanglet, I'll try to check the CI ASAP to fix it to get this merged

**Comment by VincentLanglet** [comment]
> Thank you @VincentLanglet, I'll try to check the CI ASAP to fix it to get this merged

Should I fix the test on this PR @theofidry ? It seems to mainly be CS issues

**Comment by theofidry** [comment]
@VincentLanglet the CS yes (I think it's related), but the test failures looks unrelated so in a different MR. Sorry I'm a bit overwhelmed IRL between visists and moving and stuff

**Comment by VincentLanglet** [comment]
> @VincentLanglet the CS yes (I think it's related), but the test failures looks unrelated so in a different MR. Sorry I'm a bit overwhelmed IRL between visists and moving and stuff

https://github.com/humbug/php-scoper/pull/1065

File: tests/AutoReview/GAE2ETest.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ class GAE2ETest extends TestCase
 
     public function test_github_actions_executes_all_the_e2e_tests(): void
     {
-        $expected = array_diff(E2ECollector::getE2ENames(), self::IGNORED_E2E_TESTS);
+        $expected = array_values(array_diff(E2ECollector::getE2ENames(), self::IGNORED_E2E_TESTS));
```
Comment (VincentLanglet) [inline tests/AutoReview/GAE2ETest.php:34]: Fix this error https://github.com/humbug/php-scoper/actions/runs/10667932194/job/29566542033?pr=1061

assertEqualsCanonicalizing seems to still care about key differences, it just doesn't care about the key order.

Comment (theofidry) [reply tests/AutoReview/GAE2ETest.php:34]: Extracted it in #1070

**Review by theofidry — APPROVED** [review]
LGTM otherwise

**Pull request merged**